src/clients/FtpGateway.cc

```diff
@@ -970,8 +970,10 @@ Ftp::Gateway::processReplyBody()
         return;
     }
 
+    const auto csize = data.readBuf->contentSize();
+
     /* Directory listings are special. They write ther own headers via the error objects */
-    if (!flags.http_header_sent && data.readBuf->contentSize() >= 0 && !flags.isdir)
+    if (!flags.http_header_sent && csize >= 0 && !flags.isdir)
```

Comment:
data.readBuf->contentSize() cannot be negative, right?

```suggestion
    if (!flags.http_header_sent && !flags.isdir)
```

Reply:
Removed.